### DISCUSSION TITLE
Explicit paths in entrypoint.sh

### DISCUSSION THREAD

**Discussion by Nander-bus** [lllyasviel/Fooocus#3475] [Q&A] [answered]
Why are the Paths `ORIGINALDIR=/content/app` and `DATADIR=/content/data` explicit paths instead of relative paths?
Why not add a `.` at the beginning of these variables(`ORIGINALDIR=./content/app` and `DATADIR=./content/data`), so the /content directory is created within the fooocus directory?

No normal Linux installation gives a non-root-user permission to create a new directory at `/content`.
I also don't see why this directory should be outside the fooocus directory and in my root, or anyone's root directory for that matter.

I'm asking instead of making this as a pull request, because I'm sure there is a reason why it is the way it is, but I don't get it.

**Comment by mashb1t** [comment, marked as answer]
> No normal Linux installation gives a non-root-user permission to create a new directory at /content.

I'm not sure if i got you correctly.
The /content/app and /content/data dirs are created by the root user in https://github.com/lllyasviel/Fooocus/blob/fd74b57f560c73a569a7c2658cff4e956ea0d149/Dockerfile#L18, then owned by the user `user` in https://github.com/lllyasviel/Fooocus/blob/fd74b57f560c73a569a7c2658cff4e956ea0d149/Dockerfile#L26.
User `user` is the owner of /content, so this is also fine.

> I also don't see why this directory should be outside the fooocus directory and in my root, or anyone's root directory for that matter.

You can move it wherever you like by adjusting the code, it can also be in /app or /var/www or any other directory. Feel free to change this to your preference.

This directory is the base of all data moved there when building the container, somewhere the files have to be moved. The Fooocus app directory is inside of this, and one can not mount effectively only data when this is the same dir and you want to also mount your app, but use separate volumes (optimized for virtual volumes with no file mounts to host, but docker volumes).

Please let me know if that answered your questions.

**Reply by Nander-bus** [reply]
I see you are referring to the Dockerfile.
I am not installing Fooocus inside a Docker Container, and for some reason I assumed that entrypoint.sh was simply how I am supposed to launch the application on Linux.

After giving the Readme.md another look, I now understand that entrypoint.sh is not intended for that, and that I should launch the application through launch.py.
If entrypoint.sh is intended to be called by the Dockerfile, which takes care of the file ownership issues, this answers my question. thanks.
